tests/drivers/uart/uart_async_api/src/test_uart_async.c

```diff
@@ -24,40 +24,50 @@ K_SEM_DEFINE(rx_buf_coherency, 0, 255);
 K_SEM_DEFINE(rx_buf_released, 0, 1);
 K_SEM_DEFINE(rx_disabled, 0, 1);
 
-ZTEST_BMEM volatile bool failed_in_isr;
-static ZTEST_BMEM const struct device *const uart_dev =
-	DEVICE_DT_GET(UART_NODE);
+static ZTEST_BMEM volatile bool failed_in_isr;
 
-static void read_abort_timeout(struct k_timer *timer);
-K_TIMER_DEFINE(read_abort_timer, read_abort_timeout, NULL);
+struct dut_data {
+	const struct device *dev;
+	const char *name;
+};
 
+static ZTEST_DMEM struct dut_data duts[] = {
+	{
+		.dev = DEVICE_DT_GET(UART_NODE),
+		.name = DT_NODE_FULL_NAME(UART_NODE),
+	},
+	/* More instances can be added here. */
+};
```

Comment:
Lets expand this to automatically find additional devices in the overlays: dut0, dut1, dut2 ... :)
```
static ZTEST_DMEM struct dut_data duts[] = {
	{
		.dev = DEVICE_DT_GET(UART_NODE),
		.name = DT_NODE_FULL_NAME(UART_NODE),
	},
#if DT_NODE_EXISTS(DT_NODELABEL(dut1))
	{
		.dev = DEVICE_DT_GET(DT_NODELABEL(dut1)),
		.name = DT_NODE_FULL_NAME(DT_NODELABEL(dut1)),
	},
#endif
#if DT_NODE_EXISTS(DT_NODELABEL(dut2))
	{
		.dev = DEVICE_DT_GET(DT_NODELABEL(dut2)),
		.name = DT_NODE_FULL_NAME(DT_NODELABEL(dut2)),
	},
#endif
        ...
};
```